tests/unit/test_charm.py

```diff
@@ -44,56 +43,40 @@ def setUp(self, *unused):
         self.harness.begin()
         self.peer_rel_id = self.harness.add_relation("mongodb", "mongodb")
 
-    @property
-    def mongodb_config_args(self) -> List[str]:
-        """Retrieves and parses config args."""
-        with open("tests/unit/data/mongodb_config_args.txt") as f:
-            config_args = f.read().splitlines()
-            config_args_formatted = ["\n" if arg == "\\n" else arg for arg in config_args]
-
-        return config_args_formatted
-
-    @property
-    def mongodb_config(self) -> str:
-        """Retrieves config from file."""
-        with open("tests/unit/data/mongodb_config.txt") as f:
-            config = f.read()
-
-        return config
-
+    @patch("charm.MongodbOperatorCharm._unit_ip")
+    @patch("charm.os.chown")
+    @patch("charm.pwd.getpwnam")
+    @patch("charm.systemd.daemon_reload")
+    @patch("charm.shutil")
     @patch("charm.MongodbOperatorCharm._add_repository")
     @patch("charm.MongodbOperatorCharm._install_apt_packages")
-    def test_mongodb_install(self, _add, _install):
+    def test_mongodb_install(self, _add, _install, shutil, daemon_reload, getpwnam, chown, _):
         """Test install calls correct functions."""
-        self.harness.charm.on.install.emit()
-        self.assertEqual(self.harness.charm.unit.status, MaintenanceStatus("installing MongoDB"))
-        _install.assert_called_once()
-        _add.assert_called_with(["mongodb-org"])
-
-    @patch_network_get(private_address="1.1.1.1")
-    def test_on_config_changed(self):
-        """Test config change event properly writes to mongo.conf file."""
-        open_mock = mock_open(read_data=self.mongodb_config)
-        with patch("builtins.open", open_mock, create=True):
-            self.harness.charm.on.config_changed.emit()
-
-        # TODO change expected output based on config options,(once config options are implemented)
-        open_mock.assert_called_with("/etc/mongod.conf", "w")
-        open_mock.return_value.write.assert_has_calls(
-            [mock.call(arg) for arg in self.mongodb_config_args]
-        )
+        with mock.patch("builtins.open"):
+            self.harness.charm.on.install.emit()
+            self.assertEqual(
+                self.harness.charm.unit.status, MaintenanceStatus("installing MongoDB")
+            )
+            _install.assert_called_once()
+            _add.assert_called_with(["mongodb-org"])
+            shutil.copy.assert_called_with(
+                "src/data/mongod.service", "/etc/systemd/system/mongod.service"
+            )
+            daemon_reload.assert_called()
+            getpwnam.assert_called()
+            chown.assert_called()
```

Comment:
These tests essentially just test the code execution, which might not be necessary as one would expect the core functionality is there. It's also harder to maintain if we ever make changes to `on_install`.

It might be better to test the following instead:


- Does the `src/data/mongod.service` file get updated correctly? It didn't add any funky newlines?
- Did `machine_ip` get inserted correctly?
- Does my code handle an invalid `src/data/mongod.service`?

Testing function calls is probably not needed.

Reply:
Yeah you're right, I just quickly edited this test. I believe we are now using unit tests to test conditions which are not tested by the integration test and the functionality of install is tested by the integration test so I believe this can be removed all together. Thoughts?